ndarray-linalg/src/qr.rs

```diff
@@ -147,9 +145,9 @@ where
     S2: DataMut<Elem = A> + DataOwned,
 {
     let av = a.slice(s![..n as isize, ..m as isize]);
-    let mut a = unsafe { ArrayBase::uninitialized((n, m)) };
-    for ((i, j), val) in a.indexed_iter_mut() {
-        *val = if i <= j { av[(i, j)] } else { A::zero() };
-    }
+    let mut a = replicate(&av);
+    Zip::indexed(&mut a).for_each(|(i, j), elt| {
+        if i > j { *elt = A::zero() }
+    });
     a
```

Comment:
This PR has an inefficiency here since zero elements are overwritten after copying (not benchmarked, in some cases probably faster/just as fast because the indexed iterator is slow).